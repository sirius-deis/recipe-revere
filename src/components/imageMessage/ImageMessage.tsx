import { FC } from "react";
import styles from './ImageMessage.module.css'

interface IImageMessage {
  src: string;
  desc: string;
  [x: string]: any;
}

const ImageMessage: FC<IImageMessage> = ({ src, desc }) => {
  return (
    <div className={styles['image-container']}>
      <img src={src} alt={desc} />
    </div>
  )
}

export default ImageMessage;