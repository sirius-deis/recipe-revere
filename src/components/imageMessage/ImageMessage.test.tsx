import { render, screen } from "@testing-library/react";
import ImageMessage from "./ImageMessage";


describe("ImageMessage component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <ImageMessage src="example.jpg" desc="Example Image" />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render image correctly", () => {
    render(
      <ImageMessage src="example.jpg" desc="Example Image" />
    );
    const img = screen.getByRole("img");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "example.jpg");
  });
});