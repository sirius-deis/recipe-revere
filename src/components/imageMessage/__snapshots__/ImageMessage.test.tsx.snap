// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageMessage component should match snapshot 1`] = `
<div>
  <div
    class="image-container"
  >
    <img
      alt="Example Image"
      src="example.jpg"
    />
  </div>
</div>
`;
